tests/tools.py

```diff
@@ -333,6 +334,23 @@ def emulate_raises_regex(self, exception_type, regex, function, *args, **kwargs)
                     exception_type, regex, str(e)))
         self.fail("No exception was thrown.  Expected exception {0} matching {1}".format(exception_type, regex))
 
+    def emulate_assertDictEqual(self, first, second, msg=None):
+        def fail(message):
+            self.fail(self._formatMessage(msg, message))
+
+        found = set()
+
+        for k in first.keys():
+            if k not in second:
+                fail("'{0}' is missing from second".format(k))
+            if first[k] != second[k]:
+                fail("'{0}' != '{1}' (key: {2})".format(first[k], second[k], k))
+            found.add(k)
+
+        for k in second.keys():
+            if k not in first:
```

Comment:
this should be 'if k not in found"

Reply:
Actually 'found' is not needed, since if we get here then all the keys in first are also in second. Will delete it.